browser/android/preferences/background_video_playback_tab_helper.cc

```diff
@@ -28,10 +32,10 @@ const char16_t k_youtube_background_playback_script[] =
     "    if (document._addEventListener === undefined) {"
     "        document._addEventListener = document.addEventListener;"
     "        document.addEventListener = function(a,b,c) {"
-    "            if(a != 'visibilitychange') {"
-    "                document._addEventListener(a,b,c);"
-    "            }"
-    "        };"
+    "           if(a != 'visibilitychange') {"
```

Comment:
what is this script even doing?

Reply:
It is handling background audio playback if we put app in background while video is playing

Reply:
I don't see why you need to keep a reference to the original function in `document._addEventListener` and it could potentially cause namespace collision problems. Also the timing of calling it in `DidFinishNavigation` does not guarantee that it will happen before any visibility change so you could end up in a place where it gets permanently set to not visible. This needs to be injected in the renderer and should not keep the original in `document._addEventListener`. You don't need to worry about calling it more than once if you inject it only at the right time.

Reply:
also there should be comments explaining what this is doing